src/components/Article-Gallery.jsx

```diff
@@ -1,22 +1,36 @@
 import axios from "axios";
+import { NavLink } from "react-router-dom";
 import { useEffect, useState } from "react";
 import ArticleCard from "./Article-card";
 
 export default function ArticleGallery() {
   const [articles, setArticles] = useState([]);
+  const [articleId, setArticleId] = useState("");
+  const [isPending, setIsPending] = useState(true);
   useEffect(() => {
     axios
       .get("https://north-star-articles.onrender.com/api/articles")
-      .then(({ data }) => setArticles(data.articles));
+      .then(({ data }) => {
+        setTimeout(() => {
+          setIsPending(false);
+          setArticles(data.articles);
+        }, 100);
+      });
   }, []);
   return (
-    <div className="gallery-wrapper">
-      <div className="inner-gallery-wrapper">
-        <div className="gallery-card-holder">
-          {articles.map((article) => {
-            return <ArticleCard article={article} key={article.title} />;
-          })}
-        </div>
+    <div className="outer-gallery-wrapper">
+      <div className="gallery-wrapper">
+        {isPending && <div>Loading</div>}
+        {articles.map((article) => {
+          return (
+            <NavLink
```

Comment:
You could use Link here instead of NavLink, they do the same thing but I would use NavLink when inside a navbar or something!